src/test/mir-opt/while-storage.rs

```diff
@@ -0,0 +1,59 @@
+// Test that we correctly generate StorageDead statements for while loop
+// conditions on all branches
+
+fn get_bool(c: bool) -> bool {
+    c
+}
+
+fn while_loop(c: bool) {
+    while get_bool(c) {
+        if get_bool(c) {
+            break;
+        }
+    }
+}
```

Comment:
Is there a "integration test" (i.e., some async-await code that was ICE-ing) that we can add for this commit?

Reply:
In general, I find tests like that are easier to "forward port" (e.g., if we were to make major alterations to MIR)

Reply:
The test would be to make MIRI check for double `StorageLive` and make sure that it's run on a large enough test suite.

Reply:
OK, thanks.